cmake/CompilerFlags.cmake

```diff
@@ -45,6 +45,10 @@ if(MSVC)
     set(CMAKE_SHARED_LINKER_FLAGS "/SUBSYSTEM:CONSOLE")
 endif()
 
+if(${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DMSGPACK_USE_CPP03")
```

Comment:
I don't recall this being an issue when I tested with ICC.  I was using version 17 through which may have better support for more modern standards.

Reply:
Yeah, I have ICC 14, which does not support enough of C++11. Do you think I should add a check on intel version ?

Reply:
Might be overkill because as then we'd need to check exactly which version we'd need.  Best to make a note of it for now.